src/main/java/gregtech/api/unification/material/registry/IMaterialRegistryManager.java

```diff
@@ -0,0 +1,96 @@
+package gregtech.api.unification.material.registry;
+
+import gregtech.api.GTValues;
+import gregtech.api.unification.material.Material;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+
+public interface IMaterialRegistryManager {
+
+    /**
+     * Create a registry for a modid. Accessible when in phase {@link Phase#PRE}.
+     *
+     * @param modid the mod id for the registry
+     * @return the registry for the mod
+     */
+    @Nonnull
+    MaterialRegistry createRegistry(@Nonnull String modid);
+
+    /**
+     * Get a mod's registry. Accessible during all phases.
+     *
+     * @param modid the modid of the mod
+     * @return the registry associated with the mod, or the GregTech registry if it does not have one
+     */
+    @Nonnull
+    MaterialRegistry getRegistry(@Nonnull String modid);
+
+    /**
+     * Get a mod's registry. Accessible during all phases.
+     *
+     * @param networkId the network ID of the registry
+     * @return the registry associated with the network ID, or the GregTech registry if it does not have one
+     */
+    @Nonnull
+    MaterialRegistry getRegistry(int networkId);
+
+    /**
+     * Accessible when in phases:
+     * <ul>
+     * <li>{@link Phase#OPEN}</li>
+     * <li>{@link Phase#CLOSED}</li>
+     * <li>{@link Phase#FROZEN}</li>
+     * </ul>
+     *
+     * @return all the Material Registries
+     */
+    @Nonnull
+    Collection<MaterialRegistry> getRegistries();
+
+    /**
+     * Accessible when in phases:
+     * <ul>
+     * <li>{@link Phase#CLOSED}</li>
+     * <li>{@link Phase#FROZEN}</li>
+     * </ul>
+     *
+     * @return all registered materials.
+     */
+    @Nonnull
+    Collection<Material> getRegisteredMaterials();
```

Comment:
Would it be useful to also have a per-registry version of this, where you can find all the materials registered to a specific registry?